main.py

```diff
@@ -1,24 +1,50 @@
-from preprocessing.dataloaders import concat_datasets, CustomDataset, CustomTemporalDataset
+from preprocessing.dataloaders import concat_datasets, SequentialSampler
+import numpy as np
+import torch
 from torch.utils.data import DataLoader
+from utils import set_seed, EarlyStopping, train, test
+import torch.optim as optim
+from models.spacial import BCINet
 import json
 
 
 def main():
     with open("params.json") as fp:
         params = json.load(fp)
 
-    dataset = concat_datasets(params["in_dir"])
-
-    dataLoader = DataLoader(dataset,
-                            batch_size=params['batch_size'],
-                            num_workers=2,
-                            shuffle=True)
+    # Check cuda availability
+    params['device'] = 'cuda' if torch.cuda.is_available() else 'cpu'
 
-    for idx, (data,label) in enumerate(dataLoader):
-        print(data.shape,label.shape)
-        print("-------------------------")
+    # Set seed for reproducibility
+    set_seed(1234)
 
+    dataset = concat_datasets(params["in_dir"])
```

Comment:
I would put everything related to the loading of the dataset into a function(s).

Reply:
Done. It is clear this way thank you :)